src/components/UserProfileSettings/AddTelegram.tsx

```diff
@@ -0,0 +1,268 @@
+import { FC, useCallback, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { CopyButton, ModalResponse } from 'common';
+import { useAccount } from 'hooks';
+import { useAppContext } from 'contexts/AppContext';
+import { walletToCAIP10 } from 'helpers/w2w';
+import { generateVerificationProof } from 'modules/rewards/utils/generateVerificationProof';
+import { useSendHandlesVerificationCode } from 'queries';
+import { appConfig } from 'config';
+
+import { Box, Link, Modal, Telegram, Text, TextInput } from 'blocks';
+import { shortenText } from 'helpers/UtilityHelper';
+
+type AddTelegramProps = {
+  modalControl: ModalResponse;
+  refetchSocialHandleStatus: () => void;
+  setErrorMessage: (errorMessage: string) => void;
+  setSuccessMessage: (successMessage: string) => void;
+};
+
+enum Steps {
+  EnterTelegram = 1,
+  VerifyId = 2,
+}
+
+const AddTelegram: FC<AddTelegramProps> = ({
+  modalControl,
+  refetchSocialHandleStatus,
+  // setErrorMessage,
+  // setSuccessMessage,
+}) => {
+  const { isOpen, onClose } = modalControl;
+  const { account, wallet } = useAccount();
+  const { handleConnectWalletAndEnableProfile } = useAppContext();
+
+  const caip10WalletAddress = walletToCAIP10({ account });
+  const [step, setStep] = useState(1);
+  const [telegramCode, setTelegramCode] = useState<string>('');
+  const { userPushSDKInstance } = useSelector((state: any) => {
+    return state.user;
+  });
+
+  const { mutate: sendVerification, isPending: isSendingVerification } = useSendHandlesVerificationCode();
+
+  const telegramValidationSchema = Yup.object({
+    telegram: Yup.string().required('Required'),
+  });
+
+  const telegramFormik = useFormik({
+    initialValues: { telegram: '' },
+    validationSchema: telegramValidationSchema,
+    onSubmit: () => {
+      handleSendVerificationCode();
+    },
+  });
+
```

Comment:
- Move the validation config out for component.
- the telegram verification doesn't works in browser